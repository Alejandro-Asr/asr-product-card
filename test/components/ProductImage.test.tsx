import React from 'react';
import renderer from 'react-test-renderer'
import { ProductCard, ProductImage } from '../../src/components';
import { product2 } from '../data/products';

describe('ProductImage', () => {

    test('debe de mostrar correctamente con la imagen personalizado ', () => {
        const wrapper = renderer.create(
            <ProductImage img="http://hola.jpg" />
        );
        
        expect( wrapper.toJSON() ).toMatchSnapshot();

    });

    test('Debe de mostrar el componente con la imagen del producto.', () => {
        const wrapper = renderer.create(
            <ProductCard product={ product2 }>
                {
                    () => (
                        <ProductImage img={ product2.img } />
                    )
                }
            </ProductCard>
        );
        
        expect( wrapper.toJSON() ).toMatchSnapshot();
    })
    
    
    
})
