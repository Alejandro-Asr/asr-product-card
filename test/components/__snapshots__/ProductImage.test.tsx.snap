// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage Debe de mostrar el componente con la imagen del producto. 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`ProductImage debe de mostrar correctamente con la imagen personalizado  1`] = `
<img
  alt="Product"
  className="productImg undefined"
  src="http://hola.jpg"
/>
`;
